README.md

```diff
@@ -25,113 +24,123 @@ Puppet Strings generates HTML documentation for Puppet extensions written in Pup
+   
+   To use YARD options with Puppet Strings, specify a `yardopts` file in the same directory in which you run `puppet strings`. Puppet Strings supports the Markdown format and automatically sets the YARD `markup` option to `markdown`.
+   
+   To see a list of available YARD options, run `yard help doc`. For details about YARD options configuration, see the [YARD docs](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md#config).
 
-The easiest way to install the `yard` gem is with Puppet itself:
+## Generating documentation with Puppet Strings
 
-For Puppet >= 4 and Puppet Enterprise 2015.2 and later:
+By default, Puppet Strings outputs documentation as HTML, or you can specify JSON or Markdown output instead.
 
-```
-$ puppet resource package yard provider=puppet_gem
-```
+Strings generates reference documentation based on the code and Strings code comments in all Puppet and Ruby source files under the `./manifests/`, `./functions/`, and `./lib/` directories.
 
-For Puppet 3.x:
+Strings outputs HTML of the reference information and the module README to the module's `./doc/` directory. This output can be rendered in any browser.
 
-```
-$ puppet resource package yard provider=gem
-```
+JSON and Markdown output include the reference documentation only. Strings sends JSON output to either STDOUT or to a file. Markdown output is written to a REFERENCE.md file in the module's main directory.
 
-For Puppet Enterprise 3.8 (Linux):
+### Generate documentation in HTML
 
-```
-GEM_HOME=/opt/puppet/lib/ruby/gems/1.9.1 puppet resource package yard provider=gem
-```
+To generate HTML documentation for a Puppet module, run Strings from that module's directory.
+
+1. Change directory into the module: `cd /modules/<MODULE_NAME>`.
+2. Run the command: `puppet strings`.
 
-For Puppet Enterprise 3.8 (Windows):
+To generate documentation for specific files or directories in a module, run the `puppet strings generate` subcommand and specify the files or directories as a space-separated list. 
 
 ```
-$env:GEM_HOME = "C:\Program Files\Puppet Labs\Puppet Enterprise\sys\ruby\lib\ruby\gems\2.0.0"
-puppet resource package yard provider=gem
+puppet strings generate first.pp second.pp
 ```
 
-### 2. Puppet Enterprise 3.8 in Linux only: Install the redcarpet gem
+To generate documentation for specific directories, run the `puppet strings generate` command and specify the directories:
 
 ```
-GEM_HOME=/opt/puppet/lib/ruby/gems/1.9.1 puppet resource package redcarpet provider=gem
+$ puppet strings generate 'modules/foo/lib/**/*.rb' 'modules/foo/manifests/**/*.pp' 'modules/foo/functions/**/*.pp' ...
 ```
 
-### 3. Install Puppet Strings
+### Generate documentation in Markdown
 
-Install the [puppet-strings](https://rubygems.org/gems/puppet-strings) gem. To ensure that Strings is installed in the right place, install the gem with Puppet as shown below.
+Strings outputs documentation in Markdown to either an `.md` file or to stdout.
```

Comment:
we may want to leave the part about .md out. It will write the Markdown to any file, .md is not assumed or automatically added...and now i'm seeing that I originally wrote this :) Also, you cannot send it to stdout anymore.